krux_boto/__init__.py

```diff
@@ -33,6 +32,21 @@
 NAME            = 'krux-boto'
 DEFAULT_REGION  = 'us-east-1'
 
+
+def get_instance_region():
+    """
+    Query the instance metadata service and return the region this instance is
+    placed in. If the metadata service can't be contacted, return a generic
+    default instead.
+    """
+    ### TODO: XXX This shouldn't get called if we're not on EC2.
+    zone = boto.utils.get_instance_metadata().get('placement', {}).get('availability-zone', None)
+    if zone is None:
+        krux.logging.get_logger('krux_boto').warn('get_instance_region failed to get the local instance region')
+        return 'us-east-1'
```

Comment:
Uh... I think this violates the Principle of Least Astonishment. I would definitely not expect this return value.

Can you explain how you foresee this being used?